### PR TITLE
Feat(nextjs): update to next 13

### PR DESCRIPTION
**Pull request by AugustinMauroy**
# Feat add

* Next 13
* Typescript support
* `src` directory
* Es-lint

# Why ?

Because next 13 is the lasted next version. And the index page have starting point for people learning nextjs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Welcome to your shiny new Codespace running Next.js! We've got everything fired
 
 You've got a blank canvas to work on from a git perspective as well. There's a single initial commit with the what you're seeing right now - where you go from here is up to you!
 
-Everything you do here is contained within this one codespace. There is no repository on GitHub yet. If and when you’re ready you can click "Publish Branch" and we’ll create your repository and push up your project. If you were just exploring then and have no further need for this code then you can simply delete your codespace and it's gone forever.
+Everything you do here is contained within this one Codespace. There is no repository on GitHub yet. If and when you’re ready you can click "Publish Branch" and we’ll create your repository and push up your project. If you were just exploring then and have no further need for this code then you can simply delete your Codespace and it's gone forever.
```
Comment (bdmac) [inline README.md:7]: Marketing guidelines call for using lowercase "codespace" when referring to an instance of a Codespace (the product) so this was correct before and needs to be reverted back. 🙇

**Review by bdmac — COMMENTED** [review]
Thanks so much for this PR! I noticed a couple of things when I ran a codespace on this PR.

1. The `.devcontainer/devcontainer.json` file needs to have its `"openFiles"` functionality updated to point to the new index.js location. It's still referencing `"pages/index.js"` which is was deleted in this PR.
2. I noticed when running the app that we now get an experience that looks pretty dramatically different than what we had on the previous template. Is that just a function of changes/enhancements in this new version?

New:
Screenshot 2023-03-01 at 9 21 50 AM

Current:
Screenshot 2023-03-01 at 9 22 08 AM

Weirdly I feel like the current one is a better example/demo than the new one since it's at least demonstrating functionality whereas the new one feels like a marketing landing page. Thoughts?

**Comment by AugustinMauroy** [comment]
The difference is that it is the demo of next.js not a demo created by the community.

File: README.md (hunk 1 above)
Comment (bdmac) [reply README.md:7]: Unless I'm missing something this still looks changed when it should not be. Thanks!

**Comment by mkwng** [comment]
Appreciate the PR! Some of this is much needed, as the current template is a bit out of date. 

I might be mistaken, but I did notice that there are some suggestions here that are different from a fresh next.js installation out of the box, particularly the use of the `src/` folder and `reactStrictMode`.

Inclusion of those might lead to slightly different behaviors relative to someone's local installation. Would you consider removing those?

**Comment by AugustinMauroy** [comment]
> Would you consider removing those?

Yes, but. Keeping them is good.

**Pull request closed without merging**